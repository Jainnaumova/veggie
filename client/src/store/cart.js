import axios from 'axios'

const GET_CART = 'GET_CART'
// const SUBMIT_CART = 'SUBMIT_CART'
const ADD_ORDER_ITEM = 'ADD_ORDER_ITEM'
const GET_UPDATE = 'GET_UPDATE'

const getCart = cart => {
  return {
    type: GET_CART,
    cart
  }
}

// const submitCart = order => {
//   return {
//     type: SUBMIT_CART,
//     // user,
//     order
//   }
// }

const updateCart = update => {
  return {
    type: GET_UPDATE,
    update
  }
}

const addOrderItem = orderInfo => {
  return {
    type: ADD_ORDER_ITEM,
    orderInfo
  }
}

export const orderItemInput = orderInfo => async dispatch => {
  try {
    const res = await axios.put('/api/orderItem', {orderInfo})
    dispatch(addOrderItem(res.data))
  } catch (err) {
    console.error(err)
  }
}

export const requestCart = id => async dispatch => {
  // try {
    const res = await axios.post('/api/users/cart', {userId: id})
    dispatch(getCart(res.data))
  // } catch (err) {
    // console.error(err)
  // }
}

export const requestCheckout = orderId => async dispatch => {
  try {
    const res = await axios.post('/api/users/checkout', {orderId})
    dispatch(getCart(res.data))
  } catch (err) {
    console.error(err)
  }
}

export const setTotal = info => async dispatch => {
  try {
    let res = await axios.put('/api/users/updateTotal', info)
    dispatch(updateCart(res.data))
  } catch (err) {
    console.log(err)
  }
}

export const setTotalSub = info => async dispatch => {
  try {
    let res = await axios.put('/api/users/totalSub', info)
    dispatch(updateCart(res.data))
  } catch (err) {
    console.log(err)
  }
}

export const removeItem = info => async dispatch => {
  try {
    console.log('remove item info is ', info)
    // let res = await axios.put('api/orderItem/remove', info);
    await axios.put('api/orderItem/remove', info)
  } catch (err) {
    console.log(err)
  }
}

const defaultCart = {}

export default function(state = defaultCart, action) {
  switch (action.type) {
    case GET_CART:
      return action.cart
    case GET_UPDATE:
      return action.update
    default:
      return state
  }
}

// fetch order with state cart (for checkout)
export const requestOrder = id => async dispatch => {
  try {
    console.log('id is ', id)
    const res = await axios.get(`/api/users/${id}/cart`)
    dispatch(getCart(res.data))
  } catch (err) {
    console.error(err)
  }
}

// clear cart
export const recieveEmptyCart = () => async dispatch => {
  try {
    dispatch(getCart({}))
  } catch (err) {
    console.error(err)
  }
}
