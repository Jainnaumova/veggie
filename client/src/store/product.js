import axios from 'axios'
/**
 * ACTION TYPES
 */
const GET_PRODUCT = 'GET_PRODUCT'
const GET_PRODUCTS = 'GET_PRODUCTS'

/**
 * INITIAL STATE
 */
const defaultState = {
  products: [],
  selectedProduct: {}
}

/**
 * ACTION CREATORS
 */
const getProduct = product => ({type: GET_PRODUCT, product})
const getProducts = products => ({type: GET_PRODUCTS, products})

/**
 * THUNK CREATORS
 */
export const requestProducts = () => async dispatch => {
  try {
    const res = await axios.get('/api/products')
    dispatch(getProducts(res.data))
  } catch (err) {
    console.error(err)
  }
}

export const requestProduct = id => async dispatch => {
  try {
    const res = await axios.get(`/api/products/${id}`)
    dispatch(getProduct(res.data))
  } catch (err) {
    console.error(err)
  }
}

/**
 * REDUCER
 */
export default function(state = defaultState, action) {
  switch (action.type) {
    case GET_PRODUCT:
      return {...state, selectedProduct: action.product}
    case GET_PRODUCTS:
      return {...state, products: action.products}
    default:
      return state
  }
}
