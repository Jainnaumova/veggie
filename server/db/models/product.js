const Sequelize = require('sequelize')
const db = require('../db')

const Product = db.define('product', {
  name: {
    type: Sequelize.STRING,
    unique: true,
    allowNull: false
  },
  description: {
    type: Sequelize.TEXT
  },
  inventory: {
    type: Sequelize.INTEGER,
    defaultValue: 0,
    validate: {
      min: 0
    }
  },
  imageUrl: {
    type: Sequelize.TEXT,
    defaultValue:
      'http://automation.crouzet.com/wp-content/themes/innovistasensors_wp-theme_crouzet-brands/assets/algolia/img/no-image-available.jpg'
  },
  price: {
    type: Sequelize.INTEGER,
    defaultValue: 0,
    validate: {
      min: 0
    }
  },
  tags: {
    type: Sequelize.ARRAY(Sequelize.STRING)
  }
})

module.exports = Product
