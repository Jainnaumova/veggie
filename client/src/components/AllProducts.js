import React, {Component} from 'react'
import {connect} from 'react-redux'
import {Link} from 'react-router-dom'
import ProductCart from './ProductCart'
import {requestProducts} from '../store/product'

class AllProducts extends Component {

  componentDidMount() {
    this.props.requestProducts()
  }

  render() {
    const currentProducts = this.props.products || []
    return currentProducts[0] ? (
      <div className="containerDiv">
        <h1>All Produce</h1>
        <div className="allProducts">
          <div className="products">
            {currentProducts.map(product => (
              <div key={product.id}>
                <Link to={`/products/${product.id}`}>
                  <ProductCart product={product} />
                </Link>
              </div>
            ))}
          </div>
        </div>
      </div>
    ) : (
      <p>There is currently no produce in the database</p>
    )
  }
}

const mapState = state => {
  return {
    products: state.product.products
  }
}

const mapDispatch = dispatch => {
  return {
    requestProducts: () => dispatch(requestProducts())
  }
}

export default connect(mapState, mapDispatch)(AllProducts)
