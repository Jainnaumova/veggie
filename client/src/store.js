import {createStore, combineReducers, applyMiddleware} from 'redux'
import {createLogger} from 'redux-logger'
import thunkMiddleware from 'redux-thunk'
import {composeWithDevTools} from 'redux-devtools-extension'
import user from './store/user'
import product from './store/product'
import cart from './store/cart'

const reducer = combineReducers({user, product, cart})
const middleware =
  process.env.NODE_ENV !== 'production'
    ? composeWithDevTools(
        applyMiddleware(thunkMiddleware, createLogger({collapsed: true}))
      )
    : composeWithDevTools(applyMiddleware(thunkMiddleware))
// applyMiddleware(thunkMiddleware, createLogger({collapsed: true}))

const store = createStore(reducer, middleware)

export default store
export * from './store/user'
export * from './store/product'
export * from './store/cart'

// import { createStore, applyMiddleware } from 'redux';
// import thunk from 'redux-thunk';
// import rootReducer from '../reducers/root_reducer';
//
// const middleware = [];
//
// middleware.push(thunk);
//
// if (process.env.NODE_ENV !== 'production') {
//   const logger = require('redux-logger');
//   middleware.push(logger());
// }
//
// const configureStore = (preloadedState = {}) => {
//   return createStore(rootReducer, preloadedState, applyMiddleware(...middleware));
// }
//
// export default configureStore;
