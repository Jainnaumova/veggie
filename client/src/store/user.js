import axios from 'axios'
import history from '../history'
import {requestCart} from '../store/cart'

/**
 * ACTION TYPES
 */
const GET_USER = 'GET_USER'
const REMOVE_USER = 'REMOVE_USER'
const EDIT_USER = 'EDIT_USER'

/**
 * INITIAL STATE
 */
const defaultUser = {}

/**
 * ACTION CREATORS
 */
const getUser = user => ({type: GET_USER, user})
const removeUser = () => ({type: REMOVE_USER})
const editUser = (newUserInfo) => ({type: EDIT_USER, newUserInfo})

/**
 * THUNK CREATORS
 */
export const me = () => async dispatch => {
  try {
    const res = await axios.get('/auth/me')
    dispatch(getUser(res.data || defaultUser))
  } catch (err) {
    console.error(err)
  }
}

export const auth = (email, password, method) => async dispatch => {
  let res
  try {
    res = await axios.post(`/auth/${method}`, {email, password})
    dispatch(requestCart(res.data.id))
  } catch (authError) {
    return dispatch(getUser({error: authError}))
  }

  try {
    dispatch(getUser(res.data))
    // history.push(location.state)
  } catch (dispatchOrHistoryErr) {
    console.error(dispatchOrHistoryErr)
  }
}

export const requestEdit = (id, name, email, address) => async dispatch => {
  try {
    const res = await axios.put('api/users/updateUser', {id, name, email, address})
    dispatch(editUser(res.data))
  } catch (err) {
    console.error(err)
  }
}

export const logout = () => async dispatch => {
  try {
    await axios.post('/auth/logout')
    dispatch(removeUser())
    history.push('/login')
  } catch (err) {
    console.error(err)
  }
}

/**
 * REDUCER
 */
export default function(state = defaultUser, action) {
  switch (action.type) {
    case GET_USER:
      return action.user
    case REMOVE_USER:
      return defaultUser
    case EDIT_USER:
      return action.newUserInfo
    default:
      return state
  }
}
