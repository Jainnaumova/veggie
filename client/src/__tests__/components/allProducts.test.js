import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';

import AllProducts from '../../components/AllProducts';
// import ProductCart from '../../components/ProductCart';

import {requestProducts} from '../../store/product'

jest.mock('../../components/ProductCart', () => 'ProductCart');

Enzyme.configure({ adapter: new Adapter() });

// tests for mockStore
const mockStore = configureStore();

const initialState = {
  products: []
}

const store = mockStore(initialState)

const products = [
  {
    name: 'Broccoli',
    imageUrl:
      'https://www.producemarketguide.com/sites/default/files/Commodities.tar/Commodities/broccoli_commodity-page.png',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam molestie nisl id egestas dignissim. Nullam turpis lorem, malesuada vitae fringilla a, pharetra maximus nibh. Sed id lectus ac mauris venenatis vulputate in nec erat. Pellentesque fringilla, ligula congue tempus molestie, ipsum nisl rhoncus ante, a sagittis sapien urna efficitur lectus. Vestibulum sollicitudin pharetra elit, a bibendum nibh feugiat id. Praesent efficitur, lorem non aliquet varius, tellus velit efficitur massa, et aliquam mi tortor at massa. Integer quis lacus ac risus hendrerit laoreet. Sed eu mauris pellentesque, pulvinar eros nec, viverra nulla. Vestibulum a ultricies ex. Integer bibendum ac enim non aliquet. Pellentesque aliquam ante vitae interdum eleifend.',
    inventory: 8,
    price: 799,
    tags: ['vegetable']
    },
  {
    name: 'Carrot',
    imageUrl:
      'https://purepng.com/public/uploads/large/purepng.com-carrotscarrotvegetablesfreshdeliciousefoodhealthycarrots-481521740717jmglq.png',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam molestie nisl id egestas dignissim. Nullam turpis lorem, malesuada vitae fringilla a, pharetra maximus nibh. Sed id lectus ac mauris venenatis vulputate in nec erat. Pellentesque fringilla, ligula congue tempus molestie, ipsum nisl rhoncus ante, a sagittis sapien urna efficitur lectus. Vestibulum sollicitudin pharetra elit, a bibendum nibh feugiat id. Praesent efficitur, lorem non aliquet varius, tellus velit efficitur massa, et aliquam mi tortor at massa. Integer quis lacus ac risus hendrerit laoreet. Sed eu mauris pellentesque, pulvinar eros nec, viverra nulla. Vestibulum a ultricies ex. Integer bibendum ac enim non aliquet. Pellentesque aliquam ante vitae interdum eleifend.',
    inventory: 6,
    price: 269,
    tags: ['vegetable']
  }
]
//
// describe('AllProducts container', () => {
//   let store, wrapper;
//
//   beforeEach(() => {
//     store = mockStore({
//       products
//     });
//     store.dispatch = jest.fn();
//     // wrapper = shallow(<Provider><AllProducts store={store} /></Provider>);
//     const MyContext = React.createContext(products);
//     // console.log(MyContext);
//     wrapper = shallow(<Provider store={MyContext}><AllProducts store={MyContext} /></Provider>);
//   });
//
//
//   it('maps state to props', () => {
//
//     expect(wrapper.props().products).toEqual(products);
//   });
//
//   it('maps requestProducts to dispatch an action', () => {
//     wrapper.props().requestProducts();
//     expect(store.dispatch).toHaveBeenCalledTimes(1);
//   });
//
// })

// tests for component

let props = {
  products: null,
  requestProducts: jest.fn()
}

const component = () => {
  return shallow(<AllProducts {...props} />);
}

describe('AllProducts', () => {
  // let store, wrapper;
  //
  // beforeEach(() => {
  //   // store = mockStore({
  //   //   products
  //   // });
  //   store.dispatch = jest.fn();
  //   wrapper = shallow(<AllProducts store={store} />);
  //   // const MyContext = React.createContext(products);
  //   // console.log(MyContext);
  //   // wrapper = shallow(<Provider context={MyContext}><AllProducts context={MyContext} /></Provider>);
  // });

  it('should match the snapshot', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <AllProducts {...props} />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render empty page with no produce', () => {
    expect(component().setProps({products: []}).find('p').at(0).text()).toEqual('There is currently no produce in the database');
  });

  it('should render correct amount of Product cards', () => {
    expect(component().setProps({ products: [{id:1},{id:2},{id:3}]}).find('ProductCart')).toHaveLength(3);
  });

  it('should receive props', () => {
    expect(component().instance().props).toEqual(props);
  });
})
