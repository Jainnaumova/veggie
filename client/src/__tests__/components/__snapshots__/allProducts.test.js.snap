// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllProducts should match the snapshot 1`] = `
<p>
  There is currently no produce in the database
</p>
`;
